src/System/System.jl

```diff
@@ -419,32 +419,32 @@ function polarize_spins!(sys::System{N}, dir) where N
     end
 end
 
-function get_scalar_buffers(sys::System{N}, numrequested) where N
+@inline function get_scalar_buffers(sys::System{N}, numrequested) where N
```

Comment:
Is `@inline` actually having a benefit here? If not, I'd prefer to remove it.

Reply:
Yes. The `view(sys.scalar_buffers,1:numrequested)` has to survive a function boundary if this isn't `@inline`, in which case it must be allocated.

Reply:
It seems you have done some benchmarks. Let's discuss this on our Zoom call.